File: etc/정렬/삽입정렬.py
def insertion_sort(arr):
    temp = 0
    for i in range(1, len(arr)):
        temp = arr[i]
        for j in range((i-1), -1, -1):
            if arr[j] > temp:
                arr[j+1] = arr[j]
            else:
                continue
            arr[j] = temp
    return arr


print(insertion_sort([5, 6, 1, 2, 4, 3]))

File: etc/정렬/삽입정렬.js
var insertionSort = function(array){
    var i = 1, j, temp;
    for (i; i< array.length; i++){
        temp = array[i]; // 기존 수 임시보관
        for (j=i-1; j>=0 && temp < array[j]; j--){
            array[j+1] = array[j];
        }
        array[j + 1] = temp
    }
    return array
};

insertionSort([5, 6, 1, 2, 4, 3])